src/dspbase.jl

```diff
@@ -284,12 +284,14 @@ end
 Transform the smaller convolution input to frequency domain, and return it in a
 new array. However, the contents of `buff` may be modified.
 """
-@inline function os_filter_transform!(buff::AbstractArray{<:Real}, p)
-    p * buff
+@inline function os_filter_transform!(A::NTuple{<:Any, AbstractArray{<:Real}}, p)
+    fA = (p * a for a in A)
+    .*(fA...)
 end
 
-@inline function os_filter_transform!(buff::AbstractArray{<:Complex}, p!)
-    copy(p! * buff) # p operates in place on buff
+@inline function os_filter_transform!(A::NTuple{<:Any, AbstractArray{<:Complex}}, p!)
+    fA = (copy(p! * a) for a in A) # p operates in place on buff
```

Comment:
I'll get round to this soon

Reply:
What was the reason initially for copying, by the way?

Reply:
I was trying to minimize the amount of memory allocated by the convolution routine, and did so be creating a zero-padded time domain buffer and transformation result Fourier domain buffer that I could re-use a bunch of times. Because we need to transform the filter only once, and subsequently use it to convolve each block, we need to keep the transformed filter in a separate piece of memory so it's not overwritten.

For real inputs, the Fourier transform needs a separate array to write the results to, or it creates a new array for you. In that case, we can just apply the transformation to the zero-padded filter to get a new array with the transformed filter. This new array does not share memory with either the time domain nor the fourier domain buffers. 

However, when working with complex inputs, the situation is a little different. First of all, there is only one "working" buffer instead of two, because both the time and Fourier domain are complex. This means we can store both the zero-padded signal and the transformed signal in the same buffer. Additionally, the complex Fourier transform operates in-place, which means the transform result is stored in the input array. So the filter is first put into the working buffer and zero padded, then it's transformed in that same buffer, but because we'll overwrite the working buffer's contents later with data from each block, we need to copy the transformed filter out of the working buffer into a new piece of memory so we can hang onto it.